dask/array/blockwise.py

```diff
@@ -6,7 +6,68 @@
 from .. import base, utils
 from ..delayed import unpack_collections
 from ..highlevelgraph import HighLevelGraph
-from ..blockwise import blockwise as core_blockwise
+from ..blockwise import (
+    Blockwise,
+    blockwise as core_blockwise,
+    blockwise_token,
+)
+
+
+class BlockwiseCreateArray(Blockwise):
+    """
+    Specialized Blockwise Layer for array creation routines.
+
+    Enables HighLevelGraph optimizations.
+
+    Parameters
+    ----------
+    name: string
+        The output name.
+    func : callable
+        Function to apply to populate individual blocks. This function should take
+        an iterable containing the dimensions of the given block.
+    shape: iterable
+        Iterable containing the overall shape of the array.
+    chunks: iterable
+        Iterable containing the chunk sizes along each dimension of array.
+    """
+
+    def __init__(
+        self,
+        name,
+        func,
+        shape,
+        chunks,
+    ):
+        self.name = name
```

Comment:
Is there any benefit in this? Will there ever be a case of self.name != self.output?

Reply:
No, the name is only saved here for the `__repr__` - I'd be happy to remove it if you think the parent `Blockwise.__repr__` is sufficient (perhaps we just add information about `io_deps` in that method?).

If you have the bandwidth @crusaderky, I'd be very interested to get your thoughts on #7281 (a smaller subset of this PR that should be easier to review).